vcd/resource_vcd_lb_service_monitor.go

```diff
@@ -0,0 +1,335 @@
+
+	resourceURI := strings.Split(d.Id(), ".")
+	if len(resourceURI) != 4 {
+		return nil, fmt.Errorf("resource name must be specified in such way my-org.my-org-vdc.my-edge-gw.my-lb-service-monitor")
+	}
+	orgName, vdcName, edgeName, monitorName := resourceURI[0], resourceURI[1], resourceURI[2], resourceURI[3]
+
+	vcdClient := meta.(*VCDClient)
+	edgeGateway, err := vcdClient.GetEdgeGateway(orgName, vdcName, edgeName)
+	if err != nil {
+		return nil, fmt.Errorf(errorUnableToFindEdgeGateway, err)
+	}
+
+	readLBMonitor, err := edgeGateway.ReadLBServiceMonitor(&types.LBMonitor{Name: monitorName})
+	if err != nil {
+		return []*schema.ResourceData{}, fmt.Errorf("unable to find load balancer service monitor with ID %s: %s", d.Id(), err)
+	}
+
+	d.Set("org", orgName)
+	d.Set("vdc", vdcName)
+	d.Set("edge_gateway", edgeName)
+	d.Set("name", monitorName)
+
+	d.SetId(readLBMonitor.ID)
+	return []*schema.ResourceData{d}, nil
+}
+
+// getLBMonitorType converts schema.ResourceData to *types.LBMonitor and is useful
+// for creating API requests
+func getLBMonitorType(d *schema.ResourceData) (*types.LBMonitor, error) {
+	lbMonitor := &types.LBMonitor{
+		Name:       d.Get("name").(string),
+		Interval:   d.Get("interval").(int),
+		Timeout:    d.Get("timeout").(int),
+		Type:       d.Get("type").(string),
+		MaxRetries: d.Get("max_retries").(int),
+		Expected:   d.Get("expected").(string),
+		Method:     d.Get("method").(string),
+		URL:        d.Get("url").(string),
+		Send:       d.Get("send").(string),
+		Receive:    d.Get("receive").(string),
+		Extension:  getLBMonitorExtensionType(d),
+	}
+
+	return lbMonitor, nil
+}
+
+// getLBMonitorExtensionType expands the specified map for sending via API. It appends newline to every extension as
+// per API requirement. Based on the research the underlying structure should not cause problems because duplicate keys
+// are not needed and order of the keys does not matter for API.
+// Example API call string for Extension field:
+// <extension>delay=2
+// critical=3
+// escape</extension>
+func getLBMonitorExtensionType(d *schema.ResourceData) string {
+	var extensionString string
+	extension := d.Get("extension").(map[string]interface{})
+	for k, v := range extension {
+		if k != "" && v != "" { // When key and value are given it must look like "content-type=STRING"
+			extensionString += k + "=" + v.(string) + "\n"
+		} else { // If only key is specified it does not need equals sign. Like "no-body" extension
+			extensionString += k + "\n"
+		}
+	}
+	return extensionString
+}
+
+// setLBMonitorData sets object state from *types.LBMonitor
+func setLBMonitorData(d *schema.ResourceData, lBmonitor *types.LBMonitor) error {
+	d.Set("interval", lBmonitor.Interval)
+	d.Set("timeout", lBmonitor.Timeout)
+	d.Set("max_retries", lBmonitor.MaxRetries)
+	d.Set("type", lBmonitor.Type)
+	// Optional attributes may not necessarilly
```

Comment:
"may not necessarilly" -> "may not be necessary"

Reply:
Done.